<a href="https://colab.research.google.com/github/crystalloide/Big_Data/blob/master/Installation_Docker_sous_Ubuntu_18_04.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Afficher la version du système : 


In [0]:
cat /etc/issue

Ubuntu 18.04.3 LTS \n \l



Commencez par mettre à jour votre liste de paquets existante :

Voir le lien suivant : https://www.digitalocean.com/community/tutorials/comment-installer-et-utiliser-docker-sur-ubuntu-18-04-fr

In [0]:
!sudo apt update

Hit:1 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease
Ign:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease
Ign:4 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Hit:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release
Hit:6 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release
Hit:7 http://security.ubuntu.com/ubuntu bionic-security InRelease
Hit:8 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease
Hit:11 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:12 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic InRelease
Hit:13 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:14 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Reading package lists... Done
Building dependency tree       
Readi

Ensuite, installez quelques paquets pré-requis qui permettent à ʻapt` d'utiliser les paquets via HTTPS:

In [0]:
!sudo apt install apt-transport-https ca-certificates curl software-properties-common

Reading package lists... Done
Building dependency tree       
Reading state information... Done
ca-certificates is already the newest version (20180409).
curl is already the newest version (7.58.0-2ubuntu3.8).
software-properties-common is already the newest version (0.96.24.32.12).
apt-transport-https is already the newest version (1.6.12).
0 upgraded, 0 newly installed, 0 to remove and 56 not upgraded.


Ajoutez ensuite la clé GPG du référentiel Docker officiel à votre système :

In [0]:
!curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -

OK


Ajoutez le référentiel Docker aux sources APT :

In [0]:
# sudo dnf config-manager --add repo=https://download.docker.com/linux/centos/docker-ce.repo

!sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"

Hit:1 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease
Ign:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease
Ign:4 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Hit:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release
Hit:6 http://security.ubuntu.com/ubuntu bionic-security InRelease
Hit:7 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release
Hit:8 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease
Hit:9 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:11 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:13 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic InRelease
Hit:14 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Reading package lists... Done


Ensuite, mettez à jour la base de données de paquets avec les paquets Docker du référentiel récemment ajouté :

In [0]:
!sudo apt update

Hit:1 https://download.docker.com/linux/ubuntu bionic InRelease
Hit:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran35/ InRelease
Ign:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease
Ign:4 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease
Hit:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release
Hit:6 http://security.ubuntu.com/ubuntu bionic-security InRelease
Hit:7 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release
Hit:8 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease
Hit:9 http://archive.ubuntu.com/ubuntu bionic InRelease
Hit:11 http://archive.ubuntu.com/ubuntu bionic-updates InRelease
Hit:13 http://ppa.launchpad.net/marutter/c2d4u3.5/ubuntu bionic InRelease
Hit:14 http://archive.ubuntu.com/ubuntu bionic-backports InRelease
Reading package lists... Done
Building dependency tree       
Readin

Assurez-vous que vous êtes sur le point d'installer à partir du référentiel Docker au lieu du référentiel par défaut Ubuntu :

In [0]:
!apt-cache policy docker-ce

docker-ce:
  Installed: 5:19.03.8~3-0~ubuntu-bionic
  Candidate: 5:19.03.8~3-0~ubuntu-bionic
  Version table:
 *** 5:19.03.8~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
        100 /var/lib/dpkg/status
     5:19.03.7~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
     5:19.03.6~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
     5:19.03.5~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
     5:19.03.4~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
     5:19.03.3~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
     5:19.03.2~3-0~ubuntu-bionic 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages
     5:19.03.1~3-0~ubunt

Vous verrez une sortie de données comme celle-ci, même si le numéro de version de Docker peut être différent:

Output of apt-cache policy docker-ce

docker-ce:
 Installed: (none)
 Candidate: 18.03.1 ce 3-0 ubuntu
 Version table:
     18.03.1 ce 3-0 ubuntu 500
        500 https://download.docker.com/linux/ubuntu bionic/stable amd64 Packages


Notez que docker-ce n'est pas installé, mais que le candidat à l'installation provient du référentiel Docker pour Ubuntu 18.04 (bionic).

Finalement, installez Docker :

In [0]:
!sudo apt install docker-ce

Reading package lists... Done
Building dependency tree       
Reading state information... Done
docker-ce is already the newest version (5:19.03.8~3-0~ubuntu-bionic).
0 upgraded, 0 newly installed, 0 to remove and 56 not upgraded.


Docker devrait maintenant être installé, il faut le lancer : 

In [0]:
# !sudo systemctl status docker
# !sudo systemctl start docker
!sudo service docker start
!sudo systemctl enable docker
!sudo systemctl status docker

 * Starting Docker: docker
   ...done.
Synchronizing state of docker.service with SysV service script with /lib/systemd/systemd-sysv-install.
Executing: /lib/systemd/systemd-sysv-install enable docker
System has not been booted with systemd as init system (PID 1). Can't operate.


Vérification que docker est lancé :

In [0]:
!sudo service docker status

 * Docker is not running


Utiliser docker consiste à lui transmettre une chaîne d'options et de commandes suivie d'arguments. La syntaxe prend cette forme:

docker [option] [command] [arguments]

In [0]:
!docker


Usage:	docker [OPTIONS] COMMAND

A self-sufficient runtime for containers

Options:
      --config string      Location of client config files (default "/root/.docker")
  -c, --context string     Name of the context to use to connect to the daemon (overrides DOCKER_HOST env var and default context set with "docker context use")
  -D, --debug              Enable debug mode
  -H, --host list          Daemon socket(s) to connect to
  -l, --log-level string   Set the logging level ("debug"|"info"|"warn"|"error"|"fatal") (default "info")
      --tls                Use TLS; implied by --tlsverify
      --tlscacert string   Trust certs signed only by this CA (default "/root/.docker/ca.pem")
      --tlscert string     Path to TLS certificate file (default "/root/.docker/cert.pem")
      --tlskey string      Path to TLS key file (default "/root/.docker/key.pem")
      --tlsverify          Use TLS and verify the remote
  -v, --version            Print version information and quit

Management Co

Pour afficher les options disponibles pour une commande spécifique, tapez :

In [0]:
!docker docker-subcommand --help


Usage:	docker [OPTIONS] COMMAND

A self-sufficient runtime for containers

Options:
      --config string      Location of client config files (default "/root/.docker")
  -c, --context string     Name of the context to use to connect to the daemon (overrides DOCKER_HOST env var and default context set with "docker context use")
  -D, --debug              Enable debug mode
  -H, --host list          Daemon socket(s) to connect to
  -l, --log-level string   Set the logging level ("debug"|"info"|"warn"|"error"|"fatal") (default "info")
      --tls                Use TLS; implied by --tlsverify
      --tlscacert string   Trust certs signed only by this CA (default "/root/.docker/ca.pem")
      --tlscert string     Path to TLS certificate file (default "/root/.docker/cert.pem")
      --tlskey string      Path to TLS key file (default "/root/.docker/key.pem")
      --tlsverify          Use TLS and verify the remote
  -v, --version            Print version information and quit

Management Co

Pour afficher des informations sur Docker à l’échelle du système, utilisez :

In [0]:
!sudo docker info

Client:
 Debug Mode: false

Server:
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
errors pretty printing info


Utilisation des images Docker
Les conteneurs Docker sont construits à partir d'images Docker. Par défaut, Docker extrait ces images de Docker Hub, un registre Docker géré par Docker, la société à l'origine du projet Docker. Tout le monde peut héberger ses images Docker sur Docker Hub. Ainsi, la plupart des applications et des distributions Linux dont vous aurez besoin auront des images hébergées ici.

Pour vérifier si vous pouvez accéder aux images et les télécharger à partir de Docker Hub, tapez :

In [0]:
# !su -l root
!sudo docker run hello-world

# Exemple : 
# !sudo docker run -it -p 8888:8888 gcr.io/tensorflow/tensorflow

-su: cannot set terminal process group (4076): Inappropriate ioctl for device
-su: no job control in this shell
]0;root@43a3ecb2e436: ~root@43a3ecb2e436:~# 
]0;root@43a3ecb2e436: ~root@43a3ecb2e436:~# ls
]0;root@43a3ecb2e436: ~root@43a3ecb2e436:~# ls /
bin                                        etc    opt    sys
boot                                       home   proc   tensorflow-1.15.2
content                                    lib    root   tmp
datalab                                    lib32  run    tools
dev                                        lib64  sbin   usr
dlib-19.18.0-cp27-cp27mu-linux_x86_64.whl  media  srv    var
dlib-19.18.0-cp36-cp36m-linux_x86_64.whl   mnt    swift
]0;root@43a3ecb2e436: ~root@43a3ecb2e436:~# sudo systemctl status docker
System has not been booted with systemd as init system (PID 1). Can't operate.
]0;root@43a3ecb2e436: ~root@43a3ecb2e436:~# 
]0;root@43a3ecb2e436: ~root@43a3ecb2e436:~# systemctl status docker
System has not been booted with 

In [0]:
!sudo docker search ubuntu

Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?


In [0]:
!docker run --rm -it -p 8888:8888 prakashdbhagat/docker-google-colab

docker: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?.
See 'docker run --help'.
